packages/@aws-cdk/aws-sqs/lib/queue.ts

```diff
@@ -402,6 +411,14 @@ export class Queue extends QueueBase {
         };
       }
 
+      if (encryption === QueueEncryption.SQS_MANAGED) {
```

Comment:
It looks like if the user provides an `encryptionMasterKey` _and_ sets encryption to `SQS_MANAGED`
it will be silently updated to `KMS`. Can you update the logic to throw an error if the master key
is provided and encryption is set to `SQS_MANAGED`?

Reply:
This makes sense. I added the additional check.